x/config/config.go

```diff
@@ -68,39 +69,88 @@ func WrapStreamDialer(dialer transport.StreamDialer, transportConfig string) (tr
 	return dialer, nil
 }
 
-func newStreamDialerFromPart(innerDialer transport.StreamDialer, oneDialerConfig string) (transport.StreamDialer, error) {
+type StreamDialer struct {
+	transport.StreamDialer
+	Config string
```

Comment:
Let's make it clear it's not the actual config:
```suggestion
	SanitizedConfig string
```